src/dev/src/app/datagrid/detail/detail.html

```diff
@@ -0,0 +1,60 @@
+<!--
+  ~ Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+  ~ This software is released under MIT license.
+  ~ The full license information can be found in LICENSE in the root directory of this project.
+  -->
+
+<h2>Detail Pane</h2>
+
+<style>
+clr-datagrid ::ng-deep .datagrid-detail-body {
+  padding: 0 1rem;
+}
+</style>
+
+<clr-datagrid [(clrDgSelected)]="selection">
+  <clr-dg-column [clrDgField]="'id'">User ID</clr-dg-column>
+  <clr-dg-column [clrDgField]="'name'"><ng-container *clrDgHideableColumn>Name</ng-container></clr-dg-column>
+  <clr-dg-column [clrDgField]="'creation'"><ng-container *clrDgHideableColumn>Creation date</ng-container></clr-dg-column>
+  <clr-dg-column [clrDgField]="'pokemon.name'"><ng-container *clrDgHideableColumn>Pokemon</ng-container></clr-dg-column>
+  <clr-dg-column [clrDgField]="'color'"><ng-container *clrDgHideableColumn>Favorite color</ng-container></clr-dg-column>
+
+  <clr-dg-row *clrDgItems="let user of users" [clrDgItem]="user">
+    <clr-dg-cell>{{user.id}}</clr-dg-cell>
+    <clr-dg-cell>{{user.name}}</clr-dg-cell>
+    <clr-dg-cell>{{user.creation | date}}</clr-dg-cell>
+    <clr-dg-cell>{{user.pokemon.name}}</clr-dg-cell>
+    <clr-dg-cell>
+      <span class="color-square" [style.backgroundColor]="user.color"></span>
+    </clr-dg-cell>
+
+    <!-- This is here but it will not be enabled -->
+    <clr-dg-row-detail *clrIfExpanded>
+      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin in neque in ante placerat mattis id sed quam.
+      Proin rhoncus lacus et tempor dignissim. Vivamus sem quam, pellentesque aliquet suscipit eget, pellentesque sed arcu.
+      Vivamus in dui lectus. Suspendisse cursus est ac nisl imperdiet viverra.
+    </clr-dg-row-detail>
+  </clr-dg-row>
+
+  <clr-dg-detail>
```

Comment:
Do we need this wrapper component `clr-dg-detail` at all?

Reply:
It is outside so I can find it to determine if detail pane is being used. Putting it inside a template, we'd have to ask people to name it something like `<ng-template #clrDetailPane>` which seems less explicit.

Reply:
Maybe, we could use `clrIfDetail` directive to figure out whether detail pane is being used or not? 
```
<ng-template clrIfDetail> <clr-dg-detail> ... </clr-dg-detail></ng-template>
```
Or we could create a structural directive 

```
<clr-dg-detail *clrIfDetailOpen> ... </clr-dg-detail>
```
We have a similar concept in D'nd with `*clrIfDragged`. `*clrIfDragged` used to instantiate whatever is used as a custom ghost during dragging, but also we use it to check whether a user gave us their own ghost to use.

Reply:
If we want to capture the event, we'd have to desugar. However, most of the time it could be done as this

```
<clr-dg-detail *clrIfDetail="let detail">
    <clr-dg-detail-header>...</clr-dg-detail-header>
    <clr-dg-detail-body>...</clr-dg-detail-body>
  </clr-dg-detail>```